axelor-bank-payment/src/main/java/com/axelor/apps/bankpayment/service/bankreconciliation/BankReconciliationService.java

```diff
@@ -129,60 +143,73 @@ public BankReconciliationService(
+    this.bankDetailsService = bankDetailsService;
+    this.bankReconciliationLoadAFB120Service = bankReconciliationLoadAFB120Service;
+    this.bankReconciliationLoadService = bankReconciliationLoadService;
+    this.journalRepository = journalRepository;
+    this.accountRepository = accountRepository;
   }
 
-  public void generateMovesAutoAccounting(BankReconciliation bankReconciliation) {
+  public void generateMovesAutoAccounting(BankReconciliation bankReconciliation)
+      throws AxelorException {
     Context scriptContext;
     Move move;
+    int limit = 10;
+    int offset = 0;
     List<BankReconciliationLine> bankReconciliationLines =
-        bankReconciliation.getBankReconciliationLineList();
+        bankReconciliationLineRepository
+            .findByBankReconciliation(bankReconciliation)
+            .fetch(limit, offset);
 
     List<BankStatementRule> bankStatementRules;
-
-    for (BankReconciliationLine bankReconciliationLine : bankReconciliationLines) {
-      if (bankReconciliationLine.getMoveLine() != null) continue;
-      scriptContext =
-          new Context(
-              Mapper.toMap(bankReconciliationLine.getBankStatementLine()),
-              BankStatementLineAFB120.class.getClass());
-      bankStatementRules =
-          bankStatementRuleRepository
-              .all()
-              .filter(
-                  "self.ruleType = :ruleType AND self.accountManagement.interbankCodeLine = :interbankCodeLine")
-              .bind("ruleType", BankStatementRuleRepository.RULE_TYPE_ACCOUNTING_AUTO)
-              .bind(
-                  "interbankCodeLine",
-                  bankReconciliationLine.getBankStatementLine().getOperationInterbankCodeLine())
-              .fetch();
-      for (BankStatementRule bankStatementRule : bankStatementRules) {
-
-        if (Boolean.TRUE.equals(
-            new GroovyScriptHelper(scriptContext)
-                .eval(
-                    bankStatementRule
-                        .getBankStatementQuery()
-                        .getQuery()
-                        .replaceAll("%s", "\"" + bankStatementRule.getSearchLabel() + "\"")))) {
-          if (bankStatementRule.getAccountManagement().getJournal() == null) continue;
-          move = generateMove(bankReconciliationLine, bankStatementRule);
-          try {
+    while (bankReconciliationLines.size() > 0) {
+      for (BankReconciliationLine bankReconciliationLine : bankReconciliationLines) {
+        if (bankReconciliationLine.getMoveLine() != null) {
+          continue;
+        }
+        scriptContext =
+            new Context(
+                Mapper.toMap(bankReconciliationLine.getBankStatementLine()),
+                BankStatementLineAFB120.class);
+        bankStatementRules =
+            bankStatementRuleRepository
+                .all()
+                .filter(
+                    "self.ruleType = :ruleType AND self.accountManagement.interbankCodeLine = :interbankCodeLine")
```

Comment:
Need to take into account the company.

Also, the field ruleType should be named ruleTypeSelect to follow our naming convention